pkg/scale/scale.go

```diff
@@ -411,8 +375,7 @@ func CheckIfCCTaskFinished(uTaskId, namespace, ccEndpoint, clusterName string) (
 	}
 	// No cc task found with this UID
 	if len(taskLists.UserTasks) == 0 {
-		//TODO initiate a new task in CC for graceful operation
-		return false, nil
+		return false, errorfactory.New(errorfactory.CruiseControlTaskNotExists{}, errors.New("cc task not found"), fmt.Sprintf("cc task id:%s", uTaskId))
```

Comment:
Besides CruiseControlTaskNotExists there is another failure condition for a task in CC: `CompletedWithError`
This can happen when the task itself fails in CC: killed by a user or some other condition.


I had a branch on my fork to address this:
https://github.com/amuraru/kafka-operator/commit/2e2a31d7a89ff1075622a3d39327d848348fdbc0#diff-86546d46c1322f1fa2a0e582db859473R382-R426
My proposal there was to read the task `Status` verbatim from CC output (see https://github.com/linkedin/cruise-control/blob/master/cruise-control/src/yaml/responses/userTaskState.yaml#L21-L24) and handle each case in operator